### DISCUSSION TITLE
Switch from project.json to package.json

### DISCUSSION THREAD

**Discussion by arthurgailes** [nrwl/nx#13106] [General]
Hello,

I'm have a project where all of my executors are in its package.json file, so I wanted to delete my project.json file and operate entirely within package.json for this pacakge. However, when I attempt to do so, I get the following error:

ENOENT: no such file or directory, open 'my_directory\\libs\\MYLIB\\project.json'

Is there an idiomatic way to resolve this?

**Comment by b3nab** [comment]
Yes, it could be helpful for many of us trying to do the same, so moving from project.json to a package.json.

My scenario is that I want a library to include a executable bin and therefore I need a package.json to do so.

**Comment by arthurgailes** [comment]
bumping

**Comment by jaysoo** [comment]
@arthurgailes @b3nab Is Is there a reason to use `package.json` over `project.json`? Nx works without `project.json` now, and read the scripts from `package.json`, but there are limitations to that approach.

The package-based approach has a tutorial here: https://nx.dev/getting-started/tutorials/package-based-repo-tutorial
